src/Orleans/Statistics/ThreadTrackingStatistic.cs

```diff
@@ -163,7 +162,7 @@ public void OnStartProcessing()
                 firstStart = false;
                 if (StatisticsCollector.CollectContextSwitchesStats)
                 {
-                    TrackContextSwitches();
+                    logger.IncrementMetric(metricName);        
```

Comment:
I don't think this is a good way to trigger logging these metrics.  In general, all we are doing here is telling the telemetry that there is a metric of this name that is being incremented, which is not the case.  We are then relying on the Orleans perf counter metric consumer to detect this metric and perform special processing on it.  This is quite kludgy.

The problem is that we don't have anything to log here, we just need to signal something else to retrieve and log the data, so this is really a signal to an framework extension point.  I suspect that DI could help.

Suggest something like:
1) Create public IThreadTrackingStatistic interface with the public surface of ThreadTrackingStatistic.
2) Move ThreadTrackingStatistic to OrleansTelemetryConsumers.Counters as an implementation of IThreadTrackingStatistic.
3) Add extension function to register ThreadTrackingStatistic type in the service collection for DI.
4 Update Orleans to get IThreadTrackingStatistic form DI.